Src/FluentAssertions/Equivalency/EquivalencyValidator.cs

```diff
@@ -62,6 +66,19 @@ private static void TrackWhatIsNeededToProvideContextToFailures(AssertionScope s
         scope.TrackComparands(comparands.Subject, comparands.Expectation);
     }
 
+    private static void AssertComparandsPointToActualObjects(Comparands comparands)
+    {
+        if (ReferenceEquals(comparands.Subject, comparands.Expectation))
+        {
+            return;
+        }
+
+        if (comparands.Subject is null || comparands.Expectation is null)
+        {
+            comparands.Subject.Should().BeSameAs(comparands.Expectation);
+        }
+    }
```

Comment:
On my phone, but I assume the CC complains about missing tests of these branches.

Reply:
Yeah, I was too quick with the review request. Trying to create a test for the first branch.